src/upgrades/mongodb_upgrade.py

```diff
@@ -419,6 +558,82 @@ def step_down_primary_and_wait_reelection(self) -> None:
                 if new_primary == old_primary:
                     raise FailedToElectNewPrimaryError()
 
+    def are_pre_upgrade_operations_config_server_successful(self):
+        """Runs pre-upgrade operations for config-server and returns True if successful."""
+        if not self.charm.is_role(Config.Role.CONFIG_SERVER):
+            return False
+
+        if not self.is_feature_compatibility_version(Config.Upgrade.FEATURE_VERSION_6):
+            logger.debug(
+                "Not all replicas have the expected feature compatibility: %s",
+                Config.Upgrade.FEATURE_VERSION_6,
+            )
+            return False
+
+        self.set_mongos_feature_compatibilty_version(Config.Upgrade.FEATURE_VERSION_6)
+
+        try:
+            self.turn_off_and_wait_for_balancer()
+        except BalancerStillRunningError:
+            logger.debug("Balancer is still running. Please try the pre-upgrade check later.")
+            return False
+
+        # The actions performed as a pre-upgrade check negatively impact cluster performance i.e.
+        # disabling the balancer. Users should NOT run the pre-upgrade-check action and never run
+        # the juju refresh command
```

Comment:
1) A disabled balancer can only improve performance 😂  
2) The balancer can be disabled because of maintenance or during backups, so maybe we should check it regularly on status upgrades without setting anything to peer relation. up to @Mehdi-Bendriss

Reply:
1. But in the long term disabling the balancer can result in uneven shards and not the most optimised use of resources? But I get what youre saying, reads and wirtes are probably more efficent since data isn't always getting moved around

2. You mean "update status"? or "status upgrades"?

Reply:
up to @Mehdi-Bendriss

Reply:
This can be done in a next PR - I think this is a good check to add. 

Given that the balancer can be disabled by operations other than upgrades and/or by the user themselves. We must ensure that this doesn't go unnoticed since we don't want to keep it disabled unless we meant to (temporarily). 

So I think it does make sense to check for the status of the balancer `_on_update_status()` (regardless of an upgrade running or not) - and if the balancer is disabled we put the charm in a Blocked or WaitingStatus: `waiting for balancer to be re-enabled`.